ggml/src/CMakeLists.txt

```diff
@@ -549,6 +549,13 @@ if (GGML_SYCL)
     file(GLOB   GGML_SOURCES_SYCL "ggml-sycl/*.cpp")
     list(APPEND GGML_SOURCES_SYCL "ggml-sycl.cpp")
 
+    find_package(DNNL)
+    message("-- DNNL found:"${DNNL_FOUND})
+    if (GGML_SYCL_TARGET STREQUAL "INTEL")
+        add_compile_definitions(GGML_SYCL_DNNL=${DNNL_FOUND})
+    else()
+        add_compile_definitions(GGML_SYCL_DNNL=0)
+    endif()
```

Comment:
OneDNN should also work on other vendors' device

Reply:
Others can validate onednn on other platforms, and create a new PR to open it.

Reply:
@airMeng We need to see how oneDNN performs on other platforms first, as the row `gemm` implementation for oneMKL interface introduced synchronization issues for us in the past that needed to be fixed. We should probably switch to oneDNN as well, but we need to ensure that it does not introduce regressions. 

@joeatodd, I agree with @luoyu-intel  here. What do you think?